### DISCUSSION TITLE
normal mode in Edit blocks popup?

### DISCUSSION THREAD

**Discussion by georgeguimaraes** [yetone/avante.nvim#549] [Q&A]
So, I select a block, and type `<leader>ae`, which gives me the popup:
![SCR-20240906-mrwd](https://github.com/user-attachments/assets/c1598fa7-6908-4f8e-beed-12ac33d3ca65)

But the popup is in Normal mode. I need to type  `i` to go into Insert mode:
![SCR-20240906-msbr](https://github.com/user-attachments/assets/3368ba4b-f9bc-45fa-b830-cb65150d7b0e)

Is this expected? Is there a way to enter Insert mode automatically?

(tks for the plugin btw!)

**Comment by chrishrb** [comment]
why don‘t we use `vim.input` here and instead `vim.api.nvim_open_win`? in my case I don‘t like the UI at all but instead want the default input field at the bottom
![image](https://github.com/user-attachments/assets/a7232383-2e9d-4088-bbab-1f701c0b30b5)

**Comment by iangrayhazzard** [comment]
This is my biggest gripe with this plugin too! Here's how I'm solving it until someone smarter than me figures it out:

```lua
keys = {
    {
        "<leader>ae", -- overrides default binding
        function()
            require("avante.api").edit()
            vim.schedule(function()
                vim.cmd("startinsert")
            end)
        end,
        mode = "v",
        desc = "avante: edit",
    },
},
```
